mkt/reviewers/utils.py

```diff
@@ -122,19 +124,33 @@ def create_note(self, action):
 
     def get_tested(self):
         """
-        Get string indicating device/browser used by reviewer to test.
+        Get string indicating devices/browsers used by reviewer to test.
         Will be automatically attached to the note body.
         """
-        devices = self.data.get('device_types')
-        browsers = self.data.get('browsers')
-
-        if devices and browsers:
-            return 'Tested on %s with %s' % (devices, browsers)
-        elif devices and not browsers:
-            return 'Tested on %s' % devices
-        elif not devices and browsers:
-            return 'Tested with %s' % browsers
-        return ''
+        tested_on_text = []
+        if not self.testedon_formset:
+            return ''
+        for form in self.testedon_formset.forms:
+            if form.cleaned_data:
+                dtype = form.cleaned_data.get('device_type', None)
+                device = form.cleaned_data.get('device', None)
+                version = form.cleaned_data.get('version', None)
+
+                if device and version:
+                    text = ('%s platform on %s with version %s' %
+                            (dtype, device, version))
+                elif device and not version:
+                    text = '%s platform on %s' % (dtype, device)
+                elif not device and version:
+                    text = '%s with version %s' % (dtype, version)
+                else:
+                    text = dtype
+                if text:
+                    tested_on_text.append(text)
+        if not len(tested_on_text):
+            return ''
+        else:
+            return 'Tested on ' + '; '.join(tested_on_text)
```

Comment:
it gets added to the end of the review message that gets stored in commbadge as text so it needs to be in English really.